AdobeStockImageAdminUi/Test/Mftf/Test/UserAccessesAdobeStockPanelTest.xml

```diff
@@ -18,30 +18,34 @@
             <group value="AdobeStockIntegration"/>
         </annotations>
         <before>
-            <actionGroup ref="LoginActionGroup" stepKey="loginGetFromGeneralFile"/>
+            <actionGroup ref="LoginAsAdmin" stepKey="loginAsAdmin"/>
+
+            <!-- Enable adobe stock -->
+            <actionGroup ref="AdminAdobeStockGoToConfigurationPageActionGroup" stepKey="goToAdobeStockConfigurationPage"/>
+            <actionGroup ref="AdminAdobeStockFillConfigurationActionGroup" stepKey="fillConfigurationFields"/>
+            <actionGroup ref="AdminSaveConfigActionGroup" stepKey="saveConfiguration"/>
         </before>
         <after>
-            <amOnPage url="admin/admin/auth/logout/" stepKey="amOnLogoutPage"/>
+            <!-- Disable adobe stock -->
+            <actionGroup ref="AdminAdobeStockGoToConfigurationPageActionGroup" stepKey="goToAdobeStockConfigurationPage"/>
+            <actionGroup ref="AdminAdobeStockFillConfigurationActionGroup" stepKey="fillConfigurationFields">
+                <!-- Empty not allowed by validation rules -->
+                <argument name="apiKey" value="WRONG"/>
+                <argument name="privateKey" value="WRONG"/>
+            </actionGroup>
+            <actionGroup ref="AdminSaveConfigActionGroup" stepKey="saveConfiguration"/>
+
+            <actionGroup ref="logout" stepKey="logout"/>
         </after>
-        <amOnPage url="{{CmsPagesPage.url}}" stepKey="amOnPagePagesGrid"/>
-        <waitForPageLoad stepKey="waitForPageLoad"/>
-        <click selector="{{CmsPagesPageActionsSection.addNewPageButton}}" stepKey="clickAddNewPage"/>
-        <click selector="{{CmsNewPagePageContentSection.header}}" stepKey="clickExpandContent"/>
-        <waitForPageLoad stepKey="waitForContentTabExpanded" />
-        <click selector="{{TinyMCESection.InsertImageIcon}}" stepKey="clickInsertImage" />
-        <waitForPageLoad stepKey="waitForImageDialogOpened" />
-        <click selector="{{MediaGallerySection.Browse}}" stepKey="clickBrowse" />
-        <waitForPageLoad stepKey="waitForGalleryOpen" />
-        <see selector="{{AdobeStockSection.slideOutPanelButton}}" userInput="Search Adobe Stock" stepKey="seeAdobeStockSearchButton" />
-        <click selector="{{AdobeStockSection.slideOutPanelButton}}" stepKey="openAdobeStockPanel" />
-        <waitForPageLoad stepKey="waitForAdobeStockPanelOpen" />
-        <see selector="{{AdobeStockSection.panelTitle}}" userInput="Adobe Stock" stepKey="seeAdobeStockPanelTitle" />
-        <seeElement selector="{{AdobeStockSection.searchInput}}" stepKey="seeAdobeStockSearchInput" />
-        <seeNumberOfElements selector="{{AdobeStockSection.imagesThumbnails}}" userInput="32" stepKey="see32images"/>
+        <actionGroup ref="AdminOpenAdobeStockSearchModalActionGroup" stepKey="openModal"/>
+
+        <see selector="{{AdobeStockSection.modal}} {{AdobeStockSection.panelTitle}}" userInput="{{AdobeStockModalData.title}}" stepKey="seeAdobeStockPanelTitle" />
```

Comment:
We can move these assertions to the `AdminOpenAdobeStockSearchModalActionGroup` in order to make them reusable.
The main idea is to make some particular actions reusable with all selectors and other elements packed in the action groups. So the test consists of action groups only. More reusable action groups we have, faster tests writing we get in the future.